Update react-what-is-state.md (#1961)
enkidevs/curriculum: react/introduction/fundamentals/react-what-is-state.md

}
```

This will throw an error, however, because `this.state` is `undefined` at the moment. The error would be `Cannot read property 'x' of undefined`.

> To set the **initial state**, by giving `text` a value, you can make use of `constructor()`. This method is a special function that gets executed when the component is created (creating an instance of the `React.Component` class). We'll look at these in depth in the future workouts.

Reviewer: ```suggestion
To set the **initial state**, by giving `text` a value, you can make use of `constructor()`. This method is a special function that gets executed when the component is created (creating an instance of the `React.Component` class). We'll look at these in-depth in future workouts.
```

Reviewer 2: this is good review, we definitely don't need `'the'` there. as for `'in depth'`, I am refering to the phrase/idiom not the adjective meaning that it is correct in this case

https://ell.stackexchange.com/a/102707

}
```

> This will throw an error, however, because `this.state` is `undefined` at the moment. The error would be `Cannot read property 'x' of undefined`.

Reviewer 2: hmm, seems like this is the error i get:
![image](https://user-images.githubusercontent.com/19802557/78037035-e984b880-7362-11ea-9f12-a3e163418008.png)

```suggestion
This will throw an error, however, because `this.state` is `undefined` at the moment. The error would be `TypeError: this.state is null`.
```

## Footnotes

[1:Immutable State]
This has to do with how data flows in React. The general idea behind this concept is that data goes down while actions go up, effectively creating a circular system. This will be explained in more detail at the end of the workout.

[2:Changing state before mounting]
If you try to change the `state` before the component was mounted, React will display the value of the initial `state`.

[3:Memory Leaks]

> This is mostly due to how JavaScript works, rather than React. In JS, closures can live past the lifetime of a function, and since an interval callback needs to be kept alive by the browser to get executed, it can accidentally keep other references via its closure. Thus, memory isn't cleanep up, no matter if the React component is there or not. Here's [an article](https://reliablejavascript.com/2017/11/05/how-timer-intervals-can-leak-memory/) that provides a more in-depth reading.

Reviewer 2: ```suggestion
This is mostly due to how JavaScript works, rather than React. In JS, closures can live past the lifetime of a function, and since an interval callback needs to be kept alive by the browser to get executed, it can accidentally keep other references via its closure. Thus, memory isn't cleaned up, no matter if the React component is there or not. Here's [an article](https://reliablejavascript.com/2017/11/05/how-timer-intervals-can-leak-memory/) that provides a more in-depth reading.
```

> This has to do with how data flows in React. The general idea behind this concept is that data goes down while actions go up, effectively creating a circular system. This will be explained in more detail and the end of the workout.

Reviewer: ```suggestion
This has to do with how data flows in React. The general idea behind this concept is that data goes down while actions go up, effectively creating a circular system. This will be explained in more detail at the end of the workout.
```
